src/ast/passes/resource_analyser.cpp

```diff
@@ -479,6 +479,27 @@ void ResourceAnalyser::update_map_info(Map &map)
   auto &map_info = resources_.maps_info[map.ident];
   map_info.value_type = map.type;
   map_info.key_type = map.key_type;
+
+  auto initializer = dynamic_cast<MapInitializer *>(map.initializer);
+  map_info.max_entries = initializer && initializer->max_entries
+                             ? *initializer->max_entries
+                             : bpftrace_.config_->get(
+                                   ConfigKeyInt::max_map_keys);
```

Comment:
Per comment elsewhere, I think this new map syntax should be preferred (for non-trivial programs) and should explicitly require the number of keys in order to avoid the foot gun.